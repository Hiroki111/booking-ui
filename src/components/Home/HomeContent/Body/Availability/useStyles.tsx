import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
  schedulerContainer: {
    borderRadius: '8px',
    position: 'relative',
    [theme.breakpoints.up('md')]: {
      top: '-48px',
      zIndex: '1200',
    },
  },
  toolbarRoot: {
    '& .MuiButton-label': {
      fontSize: '1rem',
      textTransform: 'capitalize',
    },
  },
}));
