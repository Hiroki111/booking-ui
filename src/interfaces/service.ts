export interface ServiceDto {
  id: number;
  name: string;
  serviceTypeId: number;
  isSelected?: boolean;
  minutes: number;
  price: number;
  tax: number;
}
