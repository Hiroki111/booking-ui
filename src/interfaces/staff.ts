import { AvailableTimeSlotDto } from './availableTimeSlot';
import { ServiceDto } from './service';

export interface StaffDto {
  id: number;
  name: string;
  title?: string;
  profilePhotoUrl?: string;
  services: ServiceDto[];
  availableDates: AvailableDate[];
}

export interface NoPreferenceStaff extends Omit<StaffDto, 'availableDates'> {
  availableDates: string[];
  isNoPreferenceStaff: boolean;
}

export interface AvailableDate {
  id: number;
  date: string;
  availableTimeSlots: AvailableTimeSlotDto[];
}
