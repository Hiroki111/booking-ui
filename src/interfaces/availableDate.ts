import { AvailableDate } from './staff';

export interface AvailableDateMap {
  [dateString: string]: AvailableDate;
}
