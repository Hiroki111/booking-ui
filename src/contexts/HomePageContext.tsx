import React, { createContext, Dispatch, useReducer, useState, useEffect } from 'react';

import { Customer } from './interfaces/customer';
import { ServiceDto } from '../interfaces/service';
import { AvailableDate, StaffDto } from '../interfaces/staff';
import { AvailableTimeSlotDto } from '../interfaces/availableTimeSlot';
import { useSessionStorage } from '../services/sessionStorage';
import { BookingState, bookingStateReducer, initialBookingState } from '../reducers/bookingState';
import { setDate, setServices, setStaff, setTimeslot, resetBookingState } from '../reducers/bookingState/actions';

interface Props {
  children: JSX.Element | JSX.Element[];
}
export interface BookingCompletionState {
  HAS_SELECTED_SERVICES: boolean;
  HAS_SELECTED_STAFF: boolean;
  HAS_SELECTED_DATE: boolean;
  HAS_SELECTED_TIMESLOT: boolean;
}
export interface HomePageContextInterface {
  serviceTypeRefs: React.RefObject<any>[];
  selectedServices: ServiceDto[];
  selectedStaff: StaffDto;
  selectedDate: AvailableDate;
  selectedTimeSlot: AvailableTimeSlotDto;
  customer: Customer;
  showBookingConfirmation: boolean;
  bookingCompletionState: BookingCompletionState;
  setServiceTypeRefs: Dispatch<React.SetStateAction<React.RefObject<any>[]>>;
  setSelectedServices: (value: ServiceDto[]) => void;
  setSelectedStaff: (value: StaffDto) => void;
  setSelectedDate: (value: AvailableDate) => void;
  setSelectedTimeSlot: (value: AvailableTimeSlotDto) => void;
  setCustomer: (value: Customer) => void;
  setShowBookingConfirmation: (value: boolean) => void;
  resetAppointmentData: () => void;
}

export const INITIAL_CUSTOMER_DATA = {
  firstName: '',
  lastName: '',
  countryPhoneCode: '',
  phoneNumber: '',
  email: '',
} as Customer;

const BOOKING_STATE_STORAGE_KEY = 'bookingState';

export const HomePageContext = createContext<HomePageContextInterface | undefined>(undefined);

export function HomePageContextProvider({ children }: Props) {
  const [serviceTypeRefs, setServiceTypeRefs] = useState<React.RefObject<any>[]>([]);
  const [bookingState, dispatchBookingState] = useReducer(
    bookingStateReducer,
    initialBookingState,
    (initialState: BookingState) =>
      JSON.parse(sessionStorage.getItem(BOOKING_STATE_STORAGE_KEY) || JSON.stringify(initialState)),
  );
  const [customer, setCustomer] = useState<Customer>(INITIAL_CUSTOMER_DATA);
  const [showBookingConfirmation, setShowBookingConfirmation] = useSessionStorage<boolean>(
    'showBookingConfirmation',
    false,
  );
  const bookingCompletionState = {
    HAS_SELECTED_SERVICES: bookingState.services?.length > 0,
    HAS_SELECTED_STAFF: !!bookingState?.staff?.id,
    HAS_SELECTED_DATE: !!bookingState?.date?.date,
    HAS_SELECTED_TIMESLOT: !!bookingState?.timeslot?.startTime,
  };

  useEffect(() => {
    sessionStorage.setItem(BOOKING_STATE_STORAGE_KEY, JSON.stringify(bookingState));
  }, [bookingState]);

  function resetAppointmentData() {
    dispatchBookingState(resetBookingState());
    setCustomer(INITIAL_CUSTOMER_DATA);
  }

  const contextValue: HomePageContextInterface = {
    serviceTypeRefs,
    selectedServices: bookingState.services,
    selectedStaff: bookingState.staff,
    selectedDate: bookingState.date,
    selectedTimeSlot: bookingState.timeslot,
    customer,
    showBookingConfirmation,
    bookingCompletionState,
    setServiceTypeRefs,
    setSelectedServices: (services: ServiceDto[]) => dispatchBookingState(setServices(services)),
    setSelectedStaff: (staff: StaffDto) => dispatchBookingState(setStaff(staff)),
    setSelectedDate: (date: AvailableDate) => dispatchBookingState(setDate(date)),
    setSelectedTimeSlot: (timeslot: AvailableTimeSlotDto) => dispatchBookingState(setTimeslot(timeslot)),
    setCustomer,
    setShowBookingConfirmation,
    resetAppointmentData,
  };

  return <HomePageContext.Provider value={contextValue}>{children}</HomePageContext.Provider>;
}

export function useHomePageContext() {
  const context = React.useContext(HomePageContext);
  if (context === undefined) {
    throw new Error('useHomePageContext must be used within HomePageContextProvider');
  }
  return context;
}
