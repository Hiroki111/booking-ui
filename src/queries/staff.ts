import { useQuery, UseQueryResult } from 'react-query';

import { FetchStaffListResponseDto, NoPreferenceStaff, StaffDto } from '../interfaces/staff';
import { ServiceDto } from '../interfaces/service';
import restApi from '../network/restApi';
import { selectRegularStaffWhoCanDoAllServices, selectAllStaffWhoCanDoAllServices } from '../services/staff';

export enum staffQuries {
  fetchStaffList = 'fetchStaffList',
}

export function useStaffQuery(): UseQueryResult<FetchStaffListResponseDto> {
  return useQuery(staffQuries.fetchStaffList, restApi.fetchStaffList);
}

export function useRegularStaffQuery(services: ServiceDto[]): UseQueryResult<StaffDto[]> {
  return useQuery(staffQuries.fetchStaffList, restApi.fetchStaffList, {
    select: (staffList: FetchStaffListResponseDto) => selectRegularStaffWhoCanDoAllServices(staffList, services),
  });
}

export function useAllStaffQuery(services: ServiceDto[]): UseQueryResult<(NoPreferenceStaff | StaffDto)[]> {
  return useQuery(staffQuries.fetchStaffList, restApi.fetchStaffList, {
    select: (staffList: FetchStaffListResponseDto) => selectAllStaffWhoCanDoAllServices(staffList, services),
  });
}
