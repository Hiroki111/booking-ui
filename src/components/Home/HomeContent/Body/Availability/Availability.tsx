import { useEffect, createRef } from 'react';
import { Redirect, useLocation } from 'react-router-dom';
import Paper from '@material-ui/core/Paper';
import { ViewState } from '@devexpress/dx-react-scheduler';
import { Scheduler, Toolbar } from '@devexpress/dx-react-scheduler-material-ui';

import { useHomePageContext } from '../../../../../contexts/HomePageContext';
import { AvailableTime } from './AvailableTime';
import { getPathToSkippedPage } from '../../../../../services/routing';
import { Calendar } from './Calendar';
import { NavigationBar } from './NavigationBar';
import { useStyles } from './useStyles';

export function Availability() {
  const classes = useStyles();
  const refToAvailableTime: React.RefObject<HTMLDivElement> = createRef();
  const { selectedDate, bookingCompletionState } = useHomePageContext();
  const location = useLocation();
  const pathToSkippedPage = getPathToSkippedPage(location.pathname, bookingCompletionState);

  useEffect(() => {
    if (selectedDate && refToAvailableTime.current) {
      refToAvailableTime.current.scrollIntoView({ behavior: 'smooth' });
    }
  }, [selectedDate, refToAvailableTime]);

  function ToolBarRootComponent(props: any) {
    return <Toolbar.Root {...props} className={classes.toolbarRoot} />;
  }

  if (pathToSkippedPage) {
    return <Redirect to={pathToSkippedPage} />;
  }

  return (
    <>
      <Paper elevation={2} className={classes.schedulerContainer}>
        <Scheduler>
          <ViewState />
          <Toolbar rootComponent={ToolBarRootComponent} />
          <NavigationBar />
          <Calendar />
        </Scheduler>
      </Paper>
      {selectedDate?.date && <AvailableTime ref={refToAvailableTime} />}
    </>
  );
}
