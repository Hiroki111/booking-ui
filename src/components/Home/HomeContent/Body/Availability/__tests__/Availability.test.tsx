import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';

import { Availability } from '../Availability';
import { HomePageContext, HomePageContextInterface } from '../../../../../../contexts/HomePageContext';
import { createMockHomePageContextValue } from '../../../../../../testUtil/mockData/HomePageContext';
import { AvailableDate } from '../../../../../../interfaces/staff';
import { createMockAvailableDate } from '../../../../../../testUtil/mockData/availableDate';
import { createMockStaff } from '../../../../../../testUtil/mockData/staff';
import { createMockServiceDto } from '../../../../../../testUtil/mockData/service';

jest.mock('../../../../../../services/routing', () => ({
  getPathToSkippedPage: jest.fn().mockImplementation(() => null),
}));

describe('Availability.tsx', () => {
  function getAvailabilityComponent(contextValue: HomePageContextInterface) {
    return (
      <MemoryRouter>
        <QueryClientProvider client={new QueryClient()}>
          <HomePageContext.Provider value={contextValue}>
            <Availability />
          </HomePageContext.Provider>
        </QueryClientProvider>
      </MemoryRouter>
    );
  }

  it('should scroll into AvailableTime when the selected date gets its value', () => {
    const mockedScrollIntoView = jest.fn();
    window.HTMLElement.prototype.scrollIntoView = mockedScrollIntoView;

    let contextValue = createMockHomePageContextValue({
      selectedDate: {} as AvailableDate,
    });
    render(getAvailabilityComponent(contextValue));
    expect(mockedScrollIntoView).not.toBeCalled();

    contextValue = createMockHomePageContextValue({
      selectedDate: createMockAvailableDate(),
    });
    render(getAvailabilityComponent(contextValue));
    expect(mockedScrollIntoView).toBeCalled();
  });

  it('should render the calendar', () => {
    jest.useFakeTimers('modern').setSystemTime(new Date('2022-03-01'));

    const service = createMockServiceDto({ minutes: 30 });
    const availableDateA = createMockAvailableDate({ date: '2022-03-01' });
    const availableDateB = createMockAvailableDate({ date: '2022-03-15' });
    const availableDateC = createMockAvailableDate({ date: '2022-03-30' });
    const staff = createMockStaff({
      services: [service],
      availableDates: [availableDateA, availableDateB, availableDateC],
    });

    const contextValue = createMockHomePageContextValue({
      selectedStaff: staff,
      selectedServices: [service],
    });
    const tree = renderer.create(getAvailabilityComponent(contextValue)).toJSON();
    // '2022-03-01', '2022-03-15', '2022-03-30' will have makeStyles-availableDay-{day} class in the snapshot
    expect(tree).toMatchSnapshot();
  });
});
