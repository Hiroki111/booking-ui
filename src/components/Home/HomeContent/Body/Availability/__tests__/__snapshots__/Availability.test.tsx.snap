// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Availability.tsx should render the calendar 1`] = `
<div
  className="MuiPaper-root makeStyles-schedulerContainer-57 MuiPaper-elevation2 MuiPaper-rounded"
>
  <div
    className="Root-container-62"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular Toolbar-toolbar-63 makeStyles-toolbarRoot-58 MuiToolbar-gutters"
    >
      <div
        className="RootBase-root-64"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root NavigationButton-button-66 makeStyles-hideNavButton-65"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiIconButton-root NavigationButton-button-66"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text OpenButton-textButton-67 makeStyles-monthLabel-59"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            March 2022
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiIconButton-root OpenButton-iconButton-68 makeStyles-monthLabel-59"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="FlexibleSpace-flexibleSpace-69"
      />
    </div>
    <div
      className="makeStyles-container-70"
      onScroll={[Function]}
    >
      <div>
        <div
          className="makeStyles-stickyElement-71 makeStyles-header-72 makeStyles-flexRow-79"
        >
          <div
            className="makeStyles-background-82 makeStyles-inlineFlex-81 makeStyles-ordinaryHeaderBorder-76"
          >
            <div>
              <table
                className="MuiTable-root makeStyles-table-84 makeStyles-table-85 makeStyles-dayScaleLayout-61"
                role={null}
              >
                <tbody
                  className="MuiTableBody-root"
                  role={null}
                >
                  <tr
                    className="MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-86"
                    >
                      <div
                        className="Cell-dayOfWeek-87"
                      >
                        Sun
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-86"
                    >
                      <div
                        className="Cell-dayOfWeek-87"
                      >
                        Mon
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-86"
                    >
                      <div
                        className="Cell-dayOfWeek-87"
                      >
                        Tue
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-86"
                    >
                      <div
                        className="Cell-dayOfWeek-87"
                      >
                        Wed
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-86"
                    >
                      <div
                        className="Cell-dayOfWeek-87"
                      >
                        Thu
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-86"
                    >
                      <div
                        className="Cell-dayOfWeek-87"
                      >
                        Fri
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-86"
                    >
                      <div
                        className="Cell-dayOfWeek-87"
                      >
                        Sat
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-flexRow-79"
        >
          <div
            className="makeStyles-inlineFlex-81"
          >
            <div
              className="makeStyles-relativeContainer-80"
            >
              <table
                className="MuiTable-root makeStyles-table-84 makeStyles-table-89 makeStyles-timeTableLayout-60"
                role={null}
              >
                <tbody
                  className="MuiTableBody-root"
                  role={null}
                >
                  <tr
                    className="MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        27
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        28
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-availableDay-91"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        Mar 1
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        2
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        3
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        4
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        5
                      </div>
                    </td>
                  </tr>
                  <tr
                    className="MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        6
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        7
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        8
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        9
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        10
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        11
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        12
                      </div>
                    </td>
                  </tr>
                  <tr
                    className="MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        13
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        14
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-availableDay-91"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        15
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        16
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        17
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        18
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        19
                      </div>
                    </td>
                  </tr>
                  <tr
                    className="MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        20
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        21
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        22
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        23
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        24
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        25
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        26
                      </div>
                    </td>
                  </tr>
                  <tr
                    className="MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        27
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        28
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        29
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-availableDay-91"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        30
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95"
                      >
                        31
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        Apr 1
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        2
                      </div>
                    </td>
                  </tr>
                  <tr
                    className="MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        3
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        4
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        5
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        6
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        7
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        8
                      </div>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body Cell-cell-94 makeStyles-dayCell-90 makeStyles-unavailableDay-92"
                      onClick={[Function]}
                      tabIndex={0}
                    >
                      <div
                        className="Cell-text-95 Cell-otherMonth-97"
                      >
                        9
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                className="AppointmentsContainer-container-101"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
